code/modules/spells/spell_types/knock.dm

```diff
@@ -11,22 +11,62 @@
 	cooldown_min = 20 //20 deciseconds reduction per rank
 
 	action_icon_state = "knock"
+	var/open_cuffs = TRUE
+
+/obj/effect/proc_holder/spell/knock/lesser
+	name = "Lesser Knock"
+	desc = "This spell opens nearby doors and closets."
+	open_cuffs = FALSE
+
+/obj/effect/proc_holder/spell/knock/choose_targets(mob/user = usr)
+	// Knock has 'targeted' and 'aoe_turf' at the same time, so it should have custom target proc.
+	var/list/targets = list()
+
+	for(var/turf/target in view_or_range(range, user, selection_type))
+		if(!can_target(target))
+			continue
+		targets += target
 
-/obj/effect/proc_holder/spell/aoe_turf/knock/cast(list/targets,mob/user = usr)
+	if(open_cuffs)
+		for(var/mob/living/carbon/target in view_or_range(range, user, selection_type))
+			if(!can_target(target))
+				continue
+			targets += target
+
+	if(!targets.len)
+		revert_cast()
+		return
```

Comment:
Since its an AOE spell, it shouldn't require a target to cast

Reply:
it is not checking 'a target'. the proc takes target list, then checks if there's anyone in the targets var. I will clarify the condition as `if(length(targets))`